Add ServiceCreateResponse.warnings

fdonze opened the pull request:
Add ServiceCreateResponse.warnings nullable field.

johnflavin commented on src/main/java/com/spotify/docker/client/messages/ServiceCreateResponse.java:
```diff
@@ -36,8 +40,21 @@
+    return create(id, null);
+  }
+  
+  static ServiceCreateResponse create(
```
Same comment as in [798](https://github.com/spotify/docker-client/pull/798#discussion_r122107311). This `create` needs to be the `@JsonCreator`, or `"Warnings"` would never be deserialized from json.

fdonze replied:
@johnflavin That's interesting, the test "testCreateServiceWithWarnings" does seem to return the "warnings" content even though the JsonCreator is not on the create with all parameters.

davidxia replied:
hm, that's strange...

johnflavin commented on src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java:
```diff
@@ -4870,6 +4870,19 @@ public void testCreateService() throws Exception {
+
+    final ServiceCreateResponse response = sut.createService(spec);
+    assertThat(response.id(), is(notNullValue()));
+    assertThat(response.warnings(), is(hasSize(1)));
```
It might be nice to also have an assertion about the contents of the warning. Like, if the warning message contains the name that wasn't found, something like `assertThat(warning.get(0), contains(whatever the random name was))`. But if you don't actually know what the warning message would be, then feel free to ignore this request.

johnflavin commented on src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java:
```diff
@@ -5552,6 +5565,21 @@ private ServiceSpec createServiceSpec(final String serviceName,
         .build();
   }
 
+  private ServiceSpec createBadServiceSpec(final String serviceName) {
```
Do you need this bit of code to be in a method rather than just part of the test? Do you plan to re-use it?

davidxia commented on src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java:
```diff
@@ -4870,6 +4870,28 @@ public void testCreateService() throws Exception {
   }
 
   @Test
+  public void testCreateServiceWithWarnings() throws Exception {
```
Could you replace this integration test with a unit test instead? See `DefaultDockerClientUnitTest` and https://github.com/spotify/docker-client/blob/master/CONTRIBUTING.md#unit-tests-and-integration-tests

fdonze replied:
@davidxia I can add a unit test. It might still be useful to keep the current test. Let me explain: I noticed testing the changes that the docker documentation is incorrect (it states that the field in "warning" and a string instead of "warnings" as an array). A unit test would not have caught this. This might not be the intent of the docker-client tests and the developer should be responsible to test the right contract but let me know if you think that the current test as any value.

davidxia replied:
Oh good catch. In that case, yes this has value. Let's keep it.

davidxia replied:
@fdonze I see the [docs here](https://docs.docker.com/engine/api/v1.25/#operation/ServiceCreate) are as you said. I wanted to create a PR for them. Where's the source code for the docs? Doesn't seem to be [here](https://github.com/moby/moby/tree/master/docs/api).

davidxia commented on src/test/java/com/spotify/docker/client/DefaultDockerClientUnitTest.java:
```diff
@@ -364,6 +369,39 @@ public void testInspectNonSwarmNode() throws Exception {
+    enqueueServerApiVersion("1.25");
+
+    server.enqueue(new MockResponse()
+            .setResponseCode(201)
```
Indentation here should be four instead of eight spaces.

fdonze commented:
@davidxia I fixed the indentation and squashed the commits. 
All tests are failing due to some strange issue: 'The command "./.travis.sh install_docker" failed and exited with 100 during ."

davidxia commented:
Don't worry about the test failures. It's caused by https://github.com/spotify/docker-client/pulls
